src/jace/translator/mapped_operation_base_translator.py

```diff
@@ -171,44 +171,44 @@ def make_input_memlets(  # noqa: PLR6301  # Subclasses might need them.
 
         # Now we will generate the input Memlets.
         tskl_inputs: dict[str, dace.Memlet] = {}
-        for i, (in_var_name, inp_shp) in enumerate(
+        for i, (in_var_name, in_shape) in enumerate(
             zip(in_var_names, (util.get_jax_var_shape(invar) for invar in eqn.invars))
         ):
-            if in_var_name is None:  # Input is a literal: No Memlet needed
-                continue
-
-            if inp_shp == ():  # Scalars
-                tskl_inputs[f"__in{i}"] = dace.Memlet.simple(in_var_name, "0")  # Scalar
-                continue
-
-            # We have to to broadcasting (combine yes and no together)
-            dims_to_bcast: Sequence[int] = [dim for dim in range(out_rank) if inp_shp[dim] == 1]
-            tskl_inputs[f"__in{i}"] = dace.Memlet.simple(
-                in_var_name,
-                ", ".join(
-                    ("0" if i in dims_to_bcast else it_var)
-                    for i, (it_var, _) in enumerate(tskl_ranges)
-                ),
-            )
+            if in_var_name is None:
+                pass
+
+            elif in_shape == ():
+                tskl_inputs[f"__in{i}"] = dace.Memlet.simple(in_var_name, "0")
+
+            else:
+                dims_to_bcast = [
+                    dim for dim in range(out_rank) if in_shape[dim] == 1 and out_shape[dim] != 1
```

Comment:
Optional suggestion:
```suggestion
                    dim for dim in range(out_rank) if in_shape[dim] == 1 != out_shape[dim]
```

Reply:
I did not know that this is possible, however, I do not think that it is very readable the `a < x < b` is useful but here I do not like it that much,